x-pack/legacy/plugins/siem/public/pages/detection_engine/rules/all_rules/columns.tsx

```diff
@@ -147,16 +147,20 @@ export const getColumns = (dispatch: React.Dispatch<Action>, kbnVersion: string)
     align: 'center',
     field: 'activate',
     name: i18n.COLUMN_ACTIVATE,
-    render: (value: TableData['activate'], item: TableData) => (
-      <RuleSwitch
-        id={item.id}
-        enabled={item.activate}
-        isLoading={item.isLoading}
-        onRuleStateChange={async (enabled, id) => {
-          await enableRulesAction([id], enabled, dispatch, kbnVersion);
-        }}
-      />
-    ),
+    render: (value: TableData['activate'], item: TableData) => {
+      const handleRuleStateChange: RuleStateChangeCallback = async (enabled, id) => {
+        await enableRulesAction([id], enabled, dispatch, kbnVersion);
+      };
```

Comment:
Nice! Thanks for the cleanup here -- after reviewing this PR I'll be better prepared next time around 🙂